code/go/0chain.net/blobbercore/allocation/deletefilechange.go

```diff
@@ -89,10 +94,12 @@ func (nf *DeleteFileChange) CommitToFileStore(ctx context.Context) error {
 						wg.Done()
 					}()
 
-					err := filestore.GetFileStore().DeleteFile(nf.AllocationID, res.ContentHash)
-					if err != nil {
-						logging.Logger.Error(fmt.Sprintf("Error while deleting file: %s", err.Error()),
-							zap.String("content_hash", res.ContentHash))
+					if count == 0 {
+						err := filestore.GetFileStore().DeleteFile(nf.AllocationID, res.ContentHash)
+						if err != nil {
+							logging.Logger.Error(fmt.Sprintf("Error while deleting file: %s", err.Error()),
+								zap.String("content_hash", res.ContentHash))
+						}
 					}
 
 					if res.ThumbnailHash != "" {
```

Comment:
I would suggest to use `res.ThumbnailSize > 0` instead of `res.ThumbnailHash != ""`

Reply:
@cnlangzi The bug you are worrying does not exist now. So I think its enough to check if thumbnail hash is empty